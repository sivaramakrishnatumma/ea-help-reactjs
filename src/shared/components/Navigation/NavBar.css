.brand-logo {
  height: 2rem;
  width: 2rem;
  margin: 0.5rem;
  vertical-align: middle;;
}

.brand-title {
  font-weight: 600;
  letter-spacing: 2px;
  vertical-align: middle;
}

.nav-item {
  border-bottom: 1px solid #4a4848;  
}

.nav-item:last-child {
  border-bottom: none;
}

.navbar-nav {
  border-top: 1px solid #4a4848; 
}

@media (min-width: 992px) {
  .nav-item {
    border-bottom: none;  
  }
  
  .navbar-nav {
    border-top: none; 
  }
}