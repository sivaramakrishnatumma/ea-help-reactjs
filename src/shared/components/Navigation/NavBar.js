import React, { useRef } from "react";
import { Link } from "react-router-dom";
import Button from "../UIElements/Button/Button";
import Image from "../UIElements/Image/Image";

import "./NavBar.css";
import NavLinks from "./NavLinks";

const NavBar = () => {
  const collapsableEl = useRef(null);

  const showMenuConent = () => {
    collapsableEl.current && collapsableEl.current.classList.toggle("collapse");
  };

  const collapseMenu = e => {
    e.preventDefault();
    collapsableEl.current && collapsableEl.current.classList.toggle("collapse");
  };

  return (
    <React.Fragment>
      <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
        <Link className="navbar-brand brand-title" to="/">
          <Image
            className="brand-logo"
            src="https://eaassets-a.akamaihd.net/eahelp/assets/83.06-RELEASE83/bundles/helpcenter/images/ea-logo.png"
          />
          HELP
        </Link>
        <Button
          className="navbar-toggler"
          type="button"
          onClick={showMenuConent}
        >
          <span className="navbar-toggler-icon"></span>
        </Button>
        <div
          ref={collapsableEl}
          className="navbar-collapse collapse"
          id="navbarNavDropdown"
        >
          <NavLinks className="collapse" collapseMenu={collapseMenu}></NavLinks>
        </div>
      </nav>
    </React.Fragment>
  );
};

export default NavBar;
