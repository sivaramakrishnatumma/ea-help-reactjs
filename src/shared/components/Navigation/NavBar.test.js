import React from "react";
import NavBar from "./NavBar";
import { shallow } from "enzyme";

describe("NavBar component test case", () => {
  it("should test NavBar component with default props", () => {
    const showMenuConent = jest.fn();
    const wrapper = shallow(<NavBar />);
    wrapper.find(".navbar-toggler").simulate("click");
    // expect(showMenuConent).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
});
