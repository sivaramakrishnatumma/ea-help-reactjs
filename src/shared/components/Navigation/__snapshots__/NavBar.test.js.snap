// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component test case should test NavBar component with default props 1`] = `
<Fragment>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <Link
      className="navbar-brand brand-title"
      to="/"
    >
      <Image
        className="brand-logo"
        src="https://eaassets-a.akamaihd.net/eahelp/assets/83.06-RELEASE83/bundles/helpcenter/images/ea-logo.png"
      />
      HELP
    </Link>
    <Button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </Button>
    <div
      className="navbar-collapse collapse"
      id="navbarNavDropdown"
    >
      <NavLinks
        className="collapse"
        collapseMenu={[Function]}
      />
    </div>
  </nav>
</Fragment>
`;
